### PR TITLE
Update changelog and increment version to v0.16

### PR DESCRIPTION
**Pull request by josh146**
* Updates the changelog

* ~Bumps the version number~ Turns out the version number had already been bumped 😆 

* Cleans up the docs, and adds links to related tutorials

### CODE REVIEW

File: strawberryfields/decompositions.py [hunk 1]
```diff
@@ -416,7 +416,7 @@ def mach_zehnder(m, n, internal_phase, external_phase, nmax):
     BS[m, n] = 1.0j / np.sqrt(2)
     BS[n, m] = 1.0j / np.sqrt(2)
     BS[n, n] = 1.0 / np.sqrt(2)
-    return BS @ Rinternal @ BS @ Rexternal
+    return np.real_if_close(BS @ Rinternal @ BS @ Rexternal)
```
Comment (josh146) [inline strawberryfields/decompositions.py:419]: Fixes a bug (?) introduced in NumPy 1.19, which results in significant complex rounding error. If not removed, this complex rounding error propagates throughout all the decompositions, producing incorrect results.

File: strawberryfields/apps/qchem/dynamics.py [hunk 2]
```diff
@@ -63,24 +63,6 @@
...
-  dynamics in molecules with a coherent input state.
-
-- The function :func:`~.sample_tmsv` generates samples for simulating vibrational quantum dynamics
-  in molecules with a two-mode squeezed vacuum input state.
-
-- The function :func:`~.prob` estimates the probability of observing a desired excitation in the
-  generated samples.
-
-- The function :func:`~.marginals` generates single-mode marginal distributions from the
-  displacement vector and covariance matrix of a Gaussian state.
```
Comment (josh146) [inline strawberryfields/apps/qchem/dynamics.py:83]: Deleted, as it duplicates the autosummary table below.

File: doc/introduction/data.rst [hunk 3]
```diff
@@ -89,6 +89,8 @@ been generated for formic acid at zero temperature. The GBS samples can be used
 
 .. autosummary::
     Formic
+    Water
+    Pyrrole
```
Comment (josh146) [inline doc/introduction/data.rst:93]: Water and Pyrrole have been added, but it looks like `PHat`, `MUTAG`, `QM9Exact`, and `QM9MC` are still missing?

File: doc/xanadu_theme/static/xanadu.css_t [hunk 4]
```diff
@@ -177,8 +177,8 @@ tt {
 }
 
 code, pre {
-  line-height: 23px;
-  margin: 20px 0;
+  line-height: 23px !important;
+  margin: 20px 0 !important;
```
Comment (josh146) [inline doc/xanadu_theme/static/xanadu.css_t:181]: A recently released version of Pygments somehow is 'hijacking' our CSS settings here

File: .github/CHANGELOG.md [hunk 5]
```diff
@@ -26,33 +26,78 @@
...
+* Updates the minimum required version of TensorFlow in the development guide.
+  [(#468)](https://github.com/XanaduAI/strawberryfields/pull/468)
 
 <h3>Contributors</h3>
 
 This release contains contributions from (in alphabetical order):
 
-Theodor Isacsson, Josh Izaac, Nathan Killoran, Fabian Laudenbach, Nicolás Quesada, Antal Száva
+Juan Miguel Arrazola, Tom Bromley, Theodor Isacsson, Josh Izaac, Soran Jahangiri, Nathan Killoran,
+Fabian Laudenbach, Nicolás Quesada, Antal Száva, ‪Ilan Tzitrin.
```
Comment (josh146) [inline .github/CHANGELOG.md:98]: Let me know if I'm missing anyone!

File: doc/introduction/data.rst (hunk 3 above)
Comment (josh146) [reply doc/introduction/data.rst:93]: @soranjh @ixfoduap @trbromley

File: strawberryfields/decompositions.py [hunk 6]
```diff
@@ -416,7 +416,7 @@ def mach_zehnder(m, n, internal_phase, external_phase, nmax):
     BS[m, n] = 1.0j / np.sqrt(2)
     BS[n, m] = 1.0j / np.sqrt(2)
     BS[n, n] = 1.0 / np.sqrt(2)
-    return BS @ Rinternal @ BS @ Rexternal
+    return np.complex128(np.real_if_close(BS @ Rinternal @ BS @ Rexternal))
```
Comment (josh146) [inline strawberryfields/decompositions.py:419]: Fixes a bug (?) introduced in NumPy 1.19, which results in significant complex rounding error. If not removed, this complex rounding error propagates throughout all the decompositions, producing incorrect results.

Comment (josh146) [reply strawberryfields/decompositions.py:419]: @nquesada @antalszava @thisac.

Note that this is roughly the same as doing
```suggestion
    return np.round(BS @ Rinternal @ BS @ Rexternal, 14)
```
based on the documentation of [`np.real_if_close()`](https://numpy.org/doc/stable/reference/generated/numpy.real_if_close.html). Should we change it to make the rounding more explicit?

File: .github/CHANGELOG.md [hunk 7]
```diff
@@ -26,33 +26,79 @@
   For more details, see the [code
   documentation](https://strawberryfields.readthedocs.io/en/stable/code/api/strawberryfields.TDMProgram.html).
 
+* Adds the function VibronicTransition to the `apps.qchem.vibronic` module. This function generates a
```
Comment (trbromley) [inline .github/CHANGELOG.md:29]:
```suggestion
* Adds the function `VibronicTransition` to the `apps.qchem.vibronic` module. This function generates a
```

File: .github/CHANGELOG.md [hunk 8]
```diff
@@ -26,33 +26,79 @@
...
+  ```pycon
+  >>> from strawberryfields.apps.qchem.vibronic import VibronicTransition
+  >>> modes = 2
+  >>> p = sf.Program(modes)
+  >>> with p.context as q:
+  ...     VibronicTransition(U1, r, U2, alpha) | q
+  ```
+
+* Adds the `TimeEvolution` function to the `apps.qchem.dynamics` module. This function
+  generates a custom sf operation for applying a time evolution operator on a given state.
```
Comment (trbromley) [inline .github/CHANGELOG.md:42]:
```suggestion
  generates a custom Strawberry Fields operation for applying a time evolution operator on a given state.
```

File: .github/CHANGELOG.md [hunk 9]
```diff
@@ -26,33 +26,79 @@
...
+  ...     sf.ops.Interferometer(Ul) | q
+  ```
+
+  where `w` is the normal mode frequencies, and `t` the time in femtoseconds.
+
+* Molecular data and pre-generated samples for water and pyrrole have been added to the data module
+  of the Applications layer of Strawberry Fields. For more details, please see the [data module
+  documentation](https://strawberryfields.readthedocs.io/en/stable/introduction/data.html#molecules)
+  [(#463)](https://github.com/XanaduAI/strawberryfields/pull/463)
+
```
Comment (trbromley) [inline .github/CHANGELOG.md:61]: Thanks for updating this!

File: doc/introduction/data.rst (hunk 3 above)
Comment (trbromley) [reply doc/introduction/data.rst:101]: Thanks @josh146! Added feature-vector based ones in e8a69683634553eb513a29946ee5ceed7dde1be4 and PHat in 5ca3df6d40a0ed9fc8ab75c71d11c33396413c65.

**Review by trbromley — APPROVED** [review]
Thanks @josh146!

File: strawberryfields/decompositions.py (hunk 6 above)
Comment (thisac) [reply strawberryfields/decompositions.py:419]: I think it's clearer what's happening with the `np.real_if_close` (despite the ugliness of nested functions like this). The `np.round` statement seems a bit more arbitrary at a glance.

**Review by thisac — APPROVED** [review]
Looks good @josh146. Thanks for fixing this!

File: .github/CHANGELOG.md [hunk 10]
```diff
@@ -26,33 +26,79 @@
...
 
-* Fixed a bug in the function `reduced_density_matrix` in `backends/tfbackend/ops.py` which caused the 
+* Fixed a bug in the function `reduced_density_matrix` in `backends/tfbackend/ops.py` which caused the
   wrong subsystems to be traced out.
   [(#467)](https://github.com/XanaduAI/strawberryfields/issues/467)
   [(#470)](https://github.com/XanaduAI/strawberryfields/pull/470)
 
+* Fixed a bug where decompositions to Mach-Zehnder interferometers would return
+  incorrect results on NumPy 1.19.
+  [(#473)](https://github.com/XanaduAI/strawberryfields/pull/473)
```
Comment (antalszava) [inline .github/CHANGELOG.md:83]: :1st_place_medal:

File: strawberryfields/decompositions.py (hunk 6 above)
Comment (antalszava) [reply strawberryfields/decompositions.py:419]: `tol=14` could be used with `np.real_if_close` (if wanted to make it more explicit)

Comment (josh146) [reply strawberryfields/decompositions.py:419]: Also, `np.round()` avoids the casting to real, and then back to complex, which adds overhead

**Review by antalszava — APPROVED** [review]
Looks good to me! :100: :blush:

**Pull request merged**